tidy3d/components/simulation.py

```diff
@@ -989,6 +991,25 @@ def _validate_monitor_size(self) -> None:
                 f"a maximum of {MAX_SIMULATION_DATA_SIZE_GB:.2f}GB are allowed."
             )
 
+    @cached_property
+    def monitors_data_size(self) -> typing.Dict:
+        """calculate simulation monitors storage size"""
+        tmesh = self.tmesh
+        grid = self.grid
+        data_size = {}
+        for monitor in self.monitors:
```

Comment:
so we already have a very similar method used in a post-init validator here
https://github.com/flexcompute/tidy3d/blob/develop/tidy3d/components/simulation.py#L959-L990

Seems like the same calculation? Rather than copy it into a new method, can we modify that `_validate_monitor_size` method to use`self.monitors_data_size` and then just iterate over the dict?

This way we only need to implement this once.

Reply:
Yeah that was the whole reason I asked for this PR, let's update the validator too @magiWei.